reframe/core/containers.py

```diff
@@ -119,32 +116,57 @@ class Sarus(ContainerPlatform):
     #: :default: :class:`False`
     with_mpi = fields.TypedField(bool)
 
+    #: Using metahub to fetch instance specific images
+    #:
+    #: :type: boolean
+    #: :default: :class:`False`
+    with_metahub = fields.TypedField(bool)
+    #: Skipping pull alltogether to speed up runs
+    #:
+    #: :type: boolean
+    #: :default: :class:`False`
+    skip_pull = fields.TypedField(bool)
+
     def __init__(self):
         super().__init__()
         self.with_mpi = False
         self._command = 'sarus'
+        self.with_metahub = False
+        self.skip_pull = False
 
     def emit_prepare_commands(self):
         # The format that Sarus uses to call the images is
         # <reposerver>/<user>/<image>:<tag>. If an image was loaded
         # locally from a tar file, the <reposerver> is 'load'.
-        if self.image.startswith('load/'):
+        if self.image.startswith('load/') or self.skip_pull:
             return []
-
-        return [self._command + ' pull %s' % self.image]
+        # Using 
+        if self.with_metahub:
+            tmpdir = tempfile.mkdtemp()
+            return [
+                'docker pull -q mh.qnib.org/'+self.image,
+                'docker save -o '+tmpdir+'/image.tar mh.qnib.org/'+self.image,
+                self._command + ' load ${TMP_DIR}/gromacs.tar mh.qnib.org/'+self.image,
+                'rm -rf '+tmpdir
+            ]
+        else:
+            return [self._command + ' pull %s' % self.image]
 
     def launch_command(self):
         super().launch_command()
         run_opts = ['--mount=type=bind,source="%s",destination="%s"' %
                     mp for mp in self.mount_points]
+        run_opts.append('--workdir='+self.workdir)
         if self.with_mpi:
             run_opts.append('--mpi')
 
         run_opts += self.options
-        run_cmd = self._command + ' run %s %s bash -c ' % (' '.join(run_opts),
+        run_cmd = self._command + ' run %s %s' % (' '.join(run_opts),
                                                            self.image)
-        return run_cmd + "'" + '; '.join(
-            ['cd ' + self.workdir] + self.commands) + "'"
+        if self.commands:
+            return run_cmd + " '" + '; '.join(self.commands) + "'"
```

Comment:
If we are not going to use `bash -c` then we don't need the single quotes nor we need the `;`. I think it's better if the user passes the whole command verbatim.